akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -213,6 +236,9 @@ object HttpEntity {
     def withContentType(contentType: ContentType): Default =
       if (contentType == this.contentType) this else copy(contentType = contentType)
 
+    def withSizeLimit(maxBytes: Long): Default =
+      copy(data = data withAttributes Attributes(SizeLimit(maxBytes, Some(contentLength))))
```

Comment:
> What's the reason not to fail early if possible?

Because someone might want to raise the limit to a higher value later.
With this PR we want to give to user the ability to override previously defined limits as often as necessary, to higher or lower values, and only have the last defined limit be applied at materialization time.